engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -311,7 +311,10 @@ public VolumeVO allocateDuplicateVolumeVO(Volume oldVol, Long templateId) {
     }
 
     private Optional<StoragePool> getPreferredStoragePool(List<StoragePool> poolList, VirtualMachine vm) {
-        String accountStoragePoolUuid = StorageManager.PreferredStoragePool.valueIn(vm.getAccountId());
+        String accountStoragePoolUuid = null;
+        if (vm != null) {
+            accountStoragePoolUuid = StorageManager.PreferredStoragePool.valueIn(vm.getAccountId());
```

Comment:
cc @nvazquez @davidjumani @sureshanaparti @weizhouapache - this needs validation, can you check what caused the regression and whether this fix actually fixes the regression or could potentially cause another one?

Reply:
The issue can be caused when a temporary snapshot (which is used only to copy a new VM or volumes) was created, I had this null pointer exception when implementing Clone a Virtual Machine https://github.com/apache/cloudstack/pull/5216/files, and this exception can be fixed after this implementation.